dll/win32/browseui/ACLCustomMRU.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * PROJECT:     ReactOS browseui
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Custom MRU AutoComplete List
+ * COPYRIGHT:   Copyright 2017 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#include "precomp.h"
+
+CACLCustomMRU::CACLCustomMRU()
+    :m_bDirty(false)
+{
+}
+
+CACLCustomMRU::~CACLCustomMRU()
+{
+    PersistMRU();
+    m_Key.Close();
+}
+
+void CACLCustomMRU::PersistMRU()
+{
+    WCHAR Key[2] = { 0, 0 };
+
+    if (!m_bDirty)
+        return;
+    m_bDirty = false;
+
+    if (m_Key.m_hKey)
+    {
+        m_Key.SetStringValue(L"MRUList", m_MRUList);
+        for (int Index = 0; Index < m_MRUList.GetLength(); ++Index)
+        {
+            Key[0] = Index + 'a';
+            m_Key.SetStringValue(Key, m_MRUData[Index]);
+        }
+    }
+}
+
+// *** IACLCustomMRU methods ***
+HRESULT STDMETHODCALLTYPE CACLCustomMRU::Initialize(LPCWSTR pwszMRURegKey, DWORD dwMax)
+{
+    LSTATUS Status = m_Key.Create(HKEY_CURRENT_USER, pwszMRURegKey);
+    if (Status != ERROR_SUCCESS)
+        return HRESULT_FROM_WIN32(Status);
+
+    m_MRUData.RemoveAll();
+    dwMax = max(0, dwMax);
+    dwMax = min(29, dwMax);
+    while (dwMax--)
+        m_MRUData.Add(CStringW());
+
+    WCHAR MRUList[40];
```

Comment:
Is there any reason for the value 40, and not more or not less?

Reply:
Because 29 items is the max, and I took a bit extra space.

Reply:
Ah ok; adding 1 short sentence comment about the 29 items max would be nie perhaps?

Reply:
4 lines above this is the `max()` call, I think that's pretty self explanatory.